python_modules/dagster/dagster/core/execution_plan/multiprocessing_engine.py

```diff
@@ -0,0 +1,152 @@
+):
+    from dagster.core.execution import yield_pipeline_execution_context
+
+    pipeline = executor_config.pipeline_fn()
+
+    inputs_to_inject = {step_key: inputs_for_step}
+
+    with yield_pipeline_execution_context(
+        pipeline, environment_dict, execution_metadata.with_tags(pid=str(os.getpid()))
+    ) as pipeline_context:
+
+        execution_plan = create_execution_plan_core(
+            pipeline_context,
+            subset_info=ExecutionPlanSubsetInfo.with_input_values([step_key], inputs_to_inject),
+        )
+
+        for step_event in start_inprocess_executor(pipeline_context, execution_plan):
+            data_to_put = {
+                'event_type': step_event.event_type,
+                'success_data': step_event.success_data,
+                'failure_data': step_event.failure_data,
+                'step_key': step_event.step.key,
+                'tags': step_event.tags,
+            }
+
+            queue.put(data_to_put)
+
+    queue.put('DONE')
+    queue.close()
+
+
+def execute_step_out_of_process(executor_config, step_context, step, prev_step_events):
+    queue = multiprocessing.Queue()
+
+    check.invariant(
+        not step_context.execution_metadata.loggers,
+        'Cannot inject loggers via ExecutionMetadata with the Multiprocess executor',
+    )
+
+    process = multiprocessing.Process(
+        target=_execute_in_child_process,
+        args=(
+            queue,
+            executor_config,
+            step_context.environment_dict,
+            step_context.execution_metadata,
+            step.key,
+            _create_input_values(step, prev_step_events),
+        ),
+    )
+
+    process.start()
+    while process.is_alive():
+        result = queue.get()
+        if result == 'DONE':
+            break
+        event_type = result['event_type']
+
+        # TODO: should we filter out? Need to think about the relationship between pipelines
+        # and subplans
+        if step.key != result['step_key']:
+            continue
+
+        check.invariant(step.key == result['step_key'])
+
+        yield ExecutionStepEvent(
+            event_type=event_type,
+            step=step,
+            success_data=result['success_data'],
+            failure_data=result['failure_data'],
+            tags=result['tags'],
+        )
+
+    # Do something reasonable on total process failure
+    process.join()
+
+
+def _all_inputs_covered(step, results):
+    for step_input in step.step_inputs:
+        if step_input.prev_output_handle not in results:
+            return False
+    return True
+
+
+def _create_input_values(step, prev_level_results):
+    input_values = {}
+    for step_input in step.step_inputs:
+        prev_output_handle = step_input.prev_output_handle
+        input_value = prev_level_results[prev_output_handle].success_data.value
+        input_values[step_input.name] = input_value
+    return input_values
+
+
+class MultiprocessExecutorConfig:
```

Comment:
can you elaborate on this -- particularly, is this where we're expecting to do things like set parallelism, etc

Reply:
yeah i'm going to work on that next. degrees of parallelism would be a property of this config object